### ISSUE TITLE
IPTV Channel skipping

### ISSUE THREAD

**Issue by antylwa** [zaps166/QMPlay2#538]
I am using IPTV by loading a m3u file. Channels are automatically skipping to the next channel during i watch a channel via 1-2 mins period. This keeps going over and over, next, next and next channel for every 2-3 minutes. I searched for the features and the opened issues but could not find a relevant issue about that problem. Do i miss something or that feature is not supported ? How can i stop channel skipping during i watch a tv channel ?

Thanks for the great work by the way.

**Comment by zaps166** [comment]
Hi, I need to reproduce it, give me steps to reproduce (including URLs, etc).

**Comment by antylwa** [comment]
Hello, im using a private iptv link. Can i send it to you privately ? Then you can try and reproduce.

**Comment by zaps166** [comment]
> Can i send it to you privately

Yes, but I don't promise that fix will be easy/possible for now and I'll look at this later :)

**Comment by TaxicabMessiah** [comment]
I agree with Realex-Fire.  I cannot watch www.toonamiaftermath.com on QMPlay2 for this reason.  The video will cut out every 60 seconds and I will have to refresh the stream to continue.

**Comment by zaps166** [comment]
Finally a stream to test! After #812 it no longer skips, but freezes for ~4 sec.

**Comment by zaps166** [comment]
But the original description is different :thinking: It plays OK to me, but sometimes video freezes for ~4 sec.

**Comment by zaps166** [comment]
> The video will cut out every 60 seconds and I will have to refresh the stream to continue.

Could you explain? I'm watching 15 minutes without issues. Short video freeze for few seconds at begin, but later plays without issues.

**Comment by zaps166** [comment]
Can't reproduce, 40 minutes and still playing correctly.

Make sure if this is enabled: ![Zrzut ekranu_20241221_183808](https://github.com/user-attachments/assets/7c0e7400-a422-4c4b-b803-2a38158ea12a)

Also you can increase this to e.g. 10 sec: ![Zrzut ekranu_20241221_183826](https://github.com/user-attachments/assets/3d443927-87a7-4d5c-8304-28609ec4f314)

---

Is it Windows with Qt6 package (it uses Schannel instead of OpenSSL)?

**Comment by zaps166** [comment]
No issues on Windows Qt5 package (which uses OpenSSL), but stops after 24 sec of playback in Qt6 package (Schannel).

**Comment by zaps166** [comment]
I'll fix Qt6 (default in Win10/11) package in next release - I'll compile FFmpeg with OpenSSL there instead of Schannel. 

Thanks for stream to test.

I forgot or didn't notice @antylwa post, sorry.

**Comment by zaps166** [comment]
Original issue report is a bit different, I probably used OpenSSL, but many things changed in meantime, so closing.

**Comment by zaps166** [comment]
If it can't resolve DNS name, FFmpeg throws end of file, so I can do nothing for now (I don't want to manually try to reconnect in this case).

---

> The video will show a couple of frames, then freeze for a while (maybe for a long time). The audio will continue to play. Is this intentional behavior for sync or a bug? When the video is unfrozen, it switches frames very quickly and then plays normally.

Yes, if it reconnects, sometimes there's a video freeze problem after d2b19c7b24f3df233f77901b5632c95ec0d848f1.

**Comment by zaps166** [comment]
> Okay, I think minor freezes associated with an unstable stream are normal 🙂

It's not minor freeze, I had ~30 sec freeze in case of ~30 sec lack of network.

**Comment by zaps166** [comment]
Fixed.

**Comment by zaps166** [comment]
Are you sure it's QMPlay2 bug, not FFmpeg bug? Are you able to check if `formatCtx` or `packet` are valid pointers? Put this before `av_read_frame`:

```c++
qDebug() << "A" << this << formatCtx << packet;
qDebug() << "B" << this << formatCtx->nb_streams;
qDebug() << "C" << this << packet->size;
```

**Comment by zaps166** [comment]
> It is likely that this is a bug in FFmpeg.

Yes, does `ffplay` crash, too?

You can try to compile ffmpeg in debug and QMPlay2 in debug with compiled ffmpeg and check again.

**Comment by zaps166** [comment]
> > Yes, does `ffplay` crash, too?
> 
> Yes, that too. Do you think QMPlay2 should catch such FFmpeg bugs so as not to crash, or is it unnecessary?

If `ffplay` crashes, it's unnecessary to use it to catch crash.

Btw. Does QMPlay2: AppImage/Windows package/Flatpak package crashes too?

**Comment by zaps166** [comment]
I don't use Wayland at all. I keep default settings for Flatpak. But when I checked on local build, it worked good on Wayland (didn't check Gnome, but Weston and Sway was OK, KDE - I don't know, because KDE doesn't launch at all on my system).

**Comment by zaps166** [comment]
Btw. Could you show screenshot with visual glitches?

**Comment by zaps166** [comment]
Just tried on Sway and Weston - not perfect, but good-enough (as for Wayland). I can't reproduce [this](https://ibb.co/DRcXHmS). Sometimes video playback goes black when opening side or bottom bar on full screen, so need to open panels again or toggle full screen.

On GNOME I see more visual glitches, but not as yours. Maybe it's GNOME issue (but it's Wayland, so any desktop implements Wayland from scratch... well...).

---

As always I'm experiencing small mouse cursor latency (I feel the latency is random depends on screen contents). Moreover on GNOME mouse cursor stutters occasionally (IMHO GNOME is most bugged Wayland implementation). I feel much better using mouse (and generally PC) with Xorg or even Windows - mouse cursor moves immediately on my screen, no limitations like can't move window or context menu is in wrong position xD

---

> Also, unlocked widgets, for some reason, do not want to move freely, etc...

Well, it Wayland... It doesn't allow you to move window ([see more](https://gitlab.xfce.org/panel-plugins/xfce4-whiskermenu-plugin/-/issues/112)) :rofl: You can move it using window manager (double click to undock and use Windows(Meta)/Alt key + mouse to move). On XWayland is works somehow (maybe it has more access to better simulate X11), but I don't know if native applications can move windows themselves (Qt can't do it maybe because of poor Wayland protocol)...

---

Maybe add `QT_QPA_PLATFORM=xcb` to `/etc/environment`? It'll "fix" all Qt applications.

---

@Realex-fire I think [this](https://github.com/zaps166/QMPlay2/discussions) is better to off-top :smile:

**Comment by zaps166** [comment]
> > If it can't resolve DNS name, FFmpeg throws end of file, so I can do nothing for now (I don't want to manually try to reconnect in this case).
> 
> Hello @zaps166, I tested the latest release on Windows 11 and the behavior is still different from Linux.
> 
> The Windows build allows only a very short network outage, whereas on Linux (AppImage) you can be disconnected for a long time. Is this intentional behavior or is the Windows build doing something wrong? You can test this e.g. by blocking outgoing connections to a specific IP address in Windows Firewall in real time.
> 
> Thanks.

Out-of-scope.

---

If now it plays normally, not 30-60 sec it's OK.
